tidy3d/components/types.py

```diff
@@ -26,71 +24,100 @@ def annotate_type(UnionType):  # pylint:disable=invalid-name
     return Annotated[UnionType, pydantic.Field(discriminator=TYPE_TAG_STR)]
 
 
+""" Numpy Arrays """
+
+
+def _totuple(arr: np.ndarray) -> tuple:
+    """Convert a numpy array to a nested tuple."""
+    try:
+        return tuple(_totuple(val) for val in arr)
+    except TypeError:
+        return arr
```

Comment:
Instead of using `try`, isn't it cleaner to use:
```python
def _to_tuple(a):
     if a.ndim > 1:
         return tuple(_to_tuple(v) for v in a)
     return tuple(a)
```

Reply:
thanks, made the change